streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -60,28 +61,57 @@
         processorContext = new StandbyContextImpl(id, config, stateMgr, metrics);
     }
 
+    @Override
+    public State state() {
+        return state;
+    }
+
+    @Override
+    public void transitionTo(final State newState) {
+        State.validateTransition(state, newState);
+        state = newState;
+    }
+
+    @Override
+    public void initializeIfNeeded() {
+        if (state() == State.CREATED) {
+            initializeMetadata();
+            initializeStateStores();
+            transitionTo(State.RUNNING);
+        }
+    }
+
     @Override
     public void initializeMetadata() {}
 
     @Override
-    public boolean initializeStateStores() {
+    public void initializeStateStores() {
         registerStateStores();
         processorContext.initialize();
         taskInitialized = true;
+    }
+
+    @Override
+    public void startRunning() {
+        // TODO: add changelog partitions to restore consumer?
+    }
+
+    @Override
+    public boolean hasChangelogs() {
         return true;
     }
 
     @Override
     public void initializeTopology() {}
 
-    /**
-     * <pre>
-     * - update offset limits
-     * </pre>
-     */
+    @Override
+    public void suspend() {
```

Comment:
Since StandbyTask do not have suspension / resumption since it should only have created / restoring / done states, I'd suggest we just remove it from Task interface and only have it in StreamTask.

Reply:
This is one way to look at it, but on the other hand, adding all interactions to the interface means that no code really has to worry about the implementation type of the task. In turn this leads to much easier testing (take a look at my work-in-progress on the TaskManagerTest). We no longer have to mock 12 classes and pre-populate them with a bunch of expected behaviors just to create a "stub" StreamTask or StandbyTask.

Reply:
I see. I was thinking that the only place we would suspend is in `listener#onPartitionsRevoked` where we only suspend active tasks, and similarly in `assignor#onAssignment` we only resume active tasks in "suspended" state, but I can buy the argument of easy testing, will make a thorough pass on TaskManager.